src/platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.cpp

```diff
@@ -0,0 +1,125 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "OTAMultiImageProcessorImpl.h"
+#include <platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.h>
+
+#include <app/clusters/ota-requestor/OTARequestorInterface.h>
+// TODO: Add this dependency when integrating the multi-OTA encryption feature
+// #include <platform/silabs/SilabsConfig.h>
+
+extern "C" {
+#include "sl_si91x_driver.h"
+}
+
+namespace chip {
+
+CHIP_ERROR OTAWiFiFirmwareProcessor::ProcessInternal(ByteSpan & block)
+{
+    sl_status_t status = SL_STATUS_OK;
+
+    ChipLogProgress(SoftwareUpdate, "Process WiFi Block");
+    // The first block is expected to be the OTA descriptor.
+    // `ProcessDescriptor` handles it and sets `mDescriptorProcessed` to avoid reprocessing.
+    if (!mDescriptorProcessed)
+    {
+        ReturnErrorOnFailure(ProcessDescriptor(block));
+#if SL_MATTER_ENABLE_OTA_ENCRYPTION
+        /* 16 bytes to used to store undecrypted data because of unalignment */
+        mAccumulator.Init(requestedOtaMaxBlockSize + 16);
+#endif // SL_MATTER_ENABLE_OTA_ENCRYPTION
+    }
+
+#if SL_MATTER_ENABLE_OTA_ENCRYPTION
+    MutableByteSpan mBlock = MutableByteSpan(mAccumulator.GetData(), mAccumulator.GetThreshold());
+    memcpy(&mBlock[0], &mBlock[requestedOtaMaxBlockSize], mUnalignmentNum);
+    memcpy(&mBlock[mUnalignmentNum], block.data(), block.size());
+
+    if (mUnalignmentNum + block.size() < requestedOtaMaxBlockSize)
+    {
+        uint32_t mAlignmentNum = (mUnalignmentNum + block.size()) / 16;
+        mAlignmentNum          = mAlignmentNum * 16;
+        mUnalignmentNum        = (mUnalignmentNum + block.size()) % 16;
+        memcpy(&mBlock[requestedOtaMaxBlockSize], &mBlock[mAlignmentNum], mUnalignmentNum);
+        mBlock.reduce_size(mAlignmentNum);
+    }
+    else
+    {
+        mUnalignmentNum = mUnalignmentNum + block.size() - requestedOtaMaxBlockSize;
+        mBlock.reduce_size(requestedOtaMaxBlockSize);
+    }
+
+    OTATlvProcessor::vOtaProcessInternalEncryption(mBlock);
+    block = mBlock;
+#endif // SL_MATTER_ENABLE_OTA_ENCRYPTION
+
+    if (mFWchunktype == SL_FWUP_RPS_HEADER)
+    {
+        // Validate block size
+        VerifyOrReturnError(block.size() >= kAlignmentBytes, CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(SoftwareUpdate, "Block too small for RPS header"));
+
+        // Use spans to reference header and content directly
+        ByteSpan rpsHeaderSpan(block.data(), kAlignmentBytes);
```

Comment:
If it possible for the block.data() size to be different from the constant? This seems like a potential size mismatch.

Reply:
I did not really understand your query. 
But here the first kAlignmentBytes(64) bytes need to be passed `sl_si91x_fwup_start` and `sl_si91x_fwup_load`. Rest of the block which is at `block.data() + kAlignmentBytes`  of size `block.size() - kAlignmentBytes` will be passing to `sl_si91x_fwup_load()`.